kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -368,4 +435,761 @@ def _document(cls):
+        """
+        # ensure phi is subsampled up to (log2_T - 1) for `phi_t * psi_f` pairs
+        max_sub_phi = lambda: max(k for k in self.phi_f if isinstance(k, int))
+        while max_sub_phi() < self.log2_T:
+            self.phi_f[max_sub_phi() + 1] = periodize_filter_fourier(
+                self.phi_f[0], nperiods=2**(max_sub_phi() + 1))
+
+    def meta(self):
+        """Get meta information on the transform
+
+        Calls the static method `compute_meta_jtfs()` with the parameters of the
+        transform object.
+
+        Returns
+        ------
+        meta : dictionary
+            See `help(kymatio.scattering1d.utils.compute_meta_jtfs)`.
+        """
+        return compute_meta_jtfs(self.J_pad, self.J, self.Q, self.J_fr, self.Q_fr,
+                                 self.T, self.F, self.aligned, self.out_3D,
+                                 self.out_type, self.resample_filters_fr,
+                                 self.average, self.sc_freq)
+
+    @property
+    def fr_attributes(self):
+        """Exposes `sc_freq`'s attributes via main object."""
+        return ('J_fr', 'Q_fr', 'shape_fr', 'shape_fr_max', 'J_pad_fr_max',
+                'average_fr', 'average_fr_global', 'aligned', 'oversampling_fr',
+                'F', 'log2_F', 'max_order_fr', 'max_pad_factor_fr', 'out_3D',
+                'resample_filters_fr', 'resample_psi_fr', 'resample_phi_fr',
+                'phi_f_fr', 'psi1_f_fr_up', 'psi1_f_fr_down')
+
+    def __getattr__(self, name):
+        # access key attributes via frequential class
+        # only called when default attribute lookup fails
+        if name in self.fr_attributes:
+            return getattr(self.sc_freq, name)
+        raise AttributeError(f"'{type(self).__name__}' object has no "
+                             f"attribute '{name}'")  # standard attribute error
+
+    # docs ###################################################################
+    @classmethod
+    def _document(cls):
+        cls.__doc__ = TimeFrequencyScatteringBase1D._doc_class.format(
+            frontend_paragraph=cls._doc_frontend_paragraph,
+            alias_call=cls._doc_alias_call,
+            parameters=cls._doc_params,
+            attributes=cls._doc_attrs,
+            sample=cls._doc_sample.format(shape=cls._doc_shape),
+        )
+        cls.scattering.__doc__ = (
+            TimeFrequencyScatteringBase1D._doc_scattering.format(
+                array=cls._doc_array,
+                n=cls._doc_array_n,
+        ))
+        # doc `sc_freq` methods
+        cls.sc_freq_compute_padding_fr.__doc__ = cls._doc_compute_padding_fr
+        cls.sc_freq_compute_J_pad.__doc__ = cls._doc_compute_J_pad
+
+    def output_size(self):
+        raise NotImplementedError("Not implemented for JTFS.")
+
+    def create_filters(self):
+        raise NotImplementedError("Implemented in `_FrequencyScatteringBase`.")
+
+    _doc_class = \
+    r"""
+    The 1D Joint Time-Frequency Scattering transform.
```

Comment:
This might be the longest docstring I have ever seen

Reply:
about on par with `Scattering1D`